lib/t/feeds.ex

```diff
@@ -97,42 +97,110 @@ defmodule T.Feeds do
         ) ::
           {[feed_profile], feed_cursor}
   def fetch_feed(user_id, location, gender, gender_preferences, count, feed_cursor) do
-    feed_profiles =
-      feed_profiles_q(user_id, gender, gender_preferences, feed_cursor)
-      |> limit(^count)
-      |> select([p], {p, distance_km(^location, p.location)})
-      |> Repo.all()
+    case feed_cursor do
+      nil -> save_eligible_feed_profiles_to_memory(user_id, location, gender, gender_preferences)
+      _ -> :ok
+    end
+
+    feed_profiles_with_cursor = feed_profiles_from_memory(user_id, feed_cursor, count)
 
     feed_cursor =
-      if last = List.last(feed_profiles) do
-        {feed_profile, _distance} = last
-        feed_profile.last_active
+      if last = List.last(feed_profiles_with_cursor) do
+        {cursor, _feed_profile} = last
+        cursor
       else
         feed_cursor
       end
 
+    feed_profiles =
+      feed_profiles_with_cursor
+      |> Enum.map(fn {_cursor, profile} ->
+        profile
+      end)
+
     {feed_profiles, feed_cursor}
   end
 
+  def save_eligible_feed_profiles_to_memory(user_id, location, gender, gender_preferences) do
+    table = String.to_atom("feed_#{user_id}")
```

Comment:
https://github.com/getsince/test3/blob/eec09d31d6fa0c55e1780ee17f8917ba080911e3/lib/t/feeds/feed_cache.ex

Reply:
1. you need to use ets as a shared resource, meaning reusing the same tables for all users, otherwise just use maps or lists, it'd be more efficient 
2. you need a process controlling the ets table which lives for as long as the app, that process owns the tables https://github.com/getsince/test3/blob/bb7811f56c826cfe1ba79d677086b13339ed2164/lib/t/feeds/feed_cache.ex#L16-L26
3. you split the tables into indexes and data, index tables need to be a tree https://github.com/getsince/test3/blob/bb7811f56c826cfe1ba79d677086b13339ed2164/lib/t/feeds/feed_cache.ex#L18, data tables can be hash tables https://github.com/getsince/test3/blob/bb7811f56c826cfe1ba79d677086b13339ed2164/lib/t/feeds/feed_cache.ex#L22-L23